File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
import java.util.*;

class Solution {
     public static Map<Integer, Character> radix = new HashMap<>() {{
        put(0, '1');
        put(1, '2');
        put(2, '4');
    }};

    public static String solution(int n) {
        String answer = "";

        while (n > 0) {
    
            n--;
            answer = radix.get(n % 3) + answer;
            n /= 3;
         
        }
        return answer;
    }
}

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<int, char> m = {{0, '1'}, {1, '2'}, {2, '4'}};

string solution(int n) {
    string answer = "";
    
    while(n > 0)
    {
        --n;
        answer = m[n % 3] + answer;
        n /= 3;
    }
    return answer;
}